### PR TITLE
Jsdoc working, method moved into subclass where logical, more doc.

### PR DESCRIPTION
**Pull request by timbl**
Also added a turtle intro designed for JS programmers

### CODE REVIEW

File: src/query.js [hunk 1]
```diff
@@ -17,7 +17,7 @@
 // Also, users name variables and want the same name back when stuff is printed
 /* jsl:option explicit*/ // Turn on JavaScriptLint variable declaration checking
 
-const IndexedFormula = require('./indexed-formula')
+const IndexedFormula = require('./store')
```
Comment (RubenVerborgh) [inline src/query.js]: I find this renaming confusing. Is it a store, an indexed formula, both, or are both the same? Seems that they are the same, but would use one term.

Comment (timbl) [reply src/query.js]: Both  "IndexedFormula" is a formula which has indexes and be queries.   A formula in N3 is a thing in the grammar which you can put in a n3 file or between two { }.  I think devs find that nomenclature .  If we change the names of these classes then we have to leave the old ones in forever more or less.  But that's ok.  I don't know how Store works with the RDFJS folks.   I think Store is just much the most logical for devs.  So we should add Store as === IndexedFormula and move toward it.

Comment (RubenVerborgh) [reply src/query.js]: It makes sense to me, but that is given my N3 background. At the moment, the mixed import is confusing:

```
const IndexedFormula = require('./store')
```

I would expect either one of these:
```
const Store = require('./store')
const IndexedFormula = require('./indexed-formula')
```

> If we change the names of these classes then we have to leave the old ones in forever more or less.

The name of the `const` is up to the one doing the require.

File: src/store.js [hunk 2]
```diff
@@ -63,8 +65,8 @@ function handleRDFType (formula, subj, pred, obj, why) {
   }
   return done // statement given is not needed if true
 }
-/**
- * IndexedFormula
+/*
+ * PanicPanic
```
Comment (RubenVerborgh) [inline src/store.js]: ? 😄

**Comment by kjetilk** [comment]
This is the solution to #269, right?

File: src/store.js (hunk 2 above)
Comment (timbl) [reply src/store.js]: That was a debug trace for trying to understand  jsdoc

Comment (timbl) [reply src/store.js]: That was a debug trace for trying to understand  jsdoc  Good catch

File: src/query.js (hunk 1 above)
Comment (timbl) [reply src/query.js]: We need people to be able to say 
```
const kb = new $rdf.Store()
or
const kb = $rdf.IndexedFormula()
```
is what I meant

Comment (RubenVerborgh) [reply src/query.js]: Just changing this in the exports should be fine then.

**Comment by timbl** [comment]
Yes, should solve https://github.com/linkeddata/rdflib.js/issues/269 in removing major brokenness... **doesn't** make the jsdoc complete.   Does allow you to point at things like  fetcher methos and updater methods and store methods

**Pull request merged**